spec/solver_spec.rb

```diff
@@ -0,0 +1,38 @@
+require './solver'
+
+describe 'Solver' do
+  context 'factorial' do
+    it 'return the factorial of the input number' do
+      number = Solver.new
+      expect(number.factorial(5)).to eq(120)
+      expect(number.factorial(0)).to eq(1)
+      expect(number.factorial(1)).to eq(1)
```

Comment:
- [ ] Kindly add tests for a negative value (`expect { resolver.factorial(-5) }.to raise_exception`)and special case `...
expect(resolver.factorial(1)).to eql(0)
...
` and ensure the tests pass